### PR TITLE
:sparkles: Improve caching 

### PR DESCRIPTION
**Pull request by varshaprasad96**
Improve performance by caching objects that only have ClusterExtension as owners.

# Description

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -123,10 +147,6 @@ func main() {
 		os.Exit(1)
 	}
 
-	if systemNamespace == "" {
-		systemNamespace = util.PodNamespace()
-	}
-
```
Comment (tmshort) [inline cmd/manager/main.go:129]: Why this?

Comment (everettraven) [reply cmd/manager/main.go:129]: Looks like it was moved higher: https://github.com/operator-framework/operator-controller/pull/825/files#diff-435eecb1d2af40ee96747f88ab71eb2758da989c92f76dcb1f714fc1b300c633R89-R91

**Comment by tmshort** [comment]
@varshaprasad96 needs rebase

**Review by tmshort — APPROVED** [review]
/approve
/lgtm

**Pull request merged**